instrumentation/opentelemetry-annotations-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/otelannotations/WithSpanTracer.java

```diff
@@ -22,6 +24,42 @@ public static WithSpanTracer tracer() {
 
   private static final Logger log = LoggerFactory.getLogger(WithSpanTracer.class);
 
+  public io.opentelemetry.api.trace.Span startSpan(
+      Context context,
+      WithSpan applicationAnnotation,
+      Method method,
+      io.opentelemetry.api.trace.Span.Kind kind) {
+
+    if (kind == io.opentelemetry.api.trace.Span.Kind.SERVER
+        && getCurrentServerSpan(context) != null) {
+      return io.opentelemetry.api.trace.Span.getInvalid();
+    }
+    if (kind == io.opentelemetry.api.trace.Span.Kind.CLIENT
+        && context.get(CONTEXT_CLIENT_SPAN_KEY) != null) {
+      return io.opentelemetry.api.trace.Span.getInvalid();
+    }
+    return startSpan(spanNameForMethodWithAnnotation(applicationAnnotation, method), kind);
+  }
+
+  /**
+   * Creates new scoped context, based on the given context, with the given span.
+   *
+   * <p>Attaches new context to the request to avoid creating duplicate server spans.
+   */
+  public Scope startScope(
+      Context context,
+      io.opentelemetry.api.trace.Span span,
+      io.opentelemetry.api.trace.Span.Kind kind) {
+
+    if (kind == io.opentelemetry.api.trace.Span.Kind.SERVER) {
+      return context.with(CONTEXT_SERVER_SPAN_KEY, span).with(span).makeCurrent();
+    }
+    if (kind == io.opentelemetry.api.trace.Span.Kind.CLIENT) {
+      return context.with(CONTEXT_CLIENT_SPAN_KEY, span).with(span).makeCurrent();
```

Comment:
See #1806

Reply:
oh this is a good point, I thought that issue was just about an optimization, but understand now 👍